### ISSUE TITLE
JSON views don't expose referenced objects properties

### ISSUE THREAD

**Issue by per-samuelsson** [Starcounter/Starcounter#3137]
Me and @jwillmer noticed that if you have this triplet:

```C#
[Database] public class Foo {
  public string Fubar;
}
[Database] public class Bar {
  public Foo Foo;
}
```

```json
{
  "Foo": { }
}
```

```C#
partial class BarPage : Json, IBound<Bar> {}
```

and you return an instance of `BarPage` like this:

```C#
var f = new Foo() { Fubar = "Baz" };
return new BarPage() { Data = new Bar() { Foo = f } };
```

the resulting model will show "Bar.Foo" as "empty".

I'm not saying this is wrong, because I have explicitly not mapped "Fubar" in the view. But it lead to confusion until @jwillmer found out the cause, and hence I want to bring it up.

1. This is by design right? And the downside of an approach where we show all public properties one level down is performance?
2. What about instead showing either `ToString()` implementation, or `GetObjectId()` as a default? I guess its because we don't have any real properties where they could be shown?
3. Do / should we support some pattern to "show all properties as key-values" for this case? Something like: `"$Default:" : "*"`.
4. Does it deserve an extra **Note** in the documentation maybe?

**Comment by chrhol** [comment]
I'm not really sure I follow what you are trying to do here.

The design is that you specify everything you want to expose in the viewmodel. The viewmodel is the master for the view, not the dataobject. So if you dont add Fubar in the viewmodel, Fubar will not be shown.

If you are trying to have some autocreated viewmodels, this is not really something we spent any time on and perhaps there is some implementation but far from complete (#654, #712)

**Comment by per-samuelsson** [comment]
> The design is that you specify everything you want to expose in the viewmodel. The viewmodel is the master for the view, not the dataobject. So if you dont add Fubar in the viewmodel, Fubar will not be shown.

Yeah, I get that. That answer my question "Is this by design". I think that approach is good.

Yet, for a fact we both got a bit confused for a while when the view model indicated "Foo" was empty. 

I think you would expect **something** to show up when you declare this

`BarPage.json`
```json
{
  "Foo": { }
}
```

and initialize a *bound* object with this:

```C#
var p = new BarPage() { Data = new Bar() };
```

Again, I'm not saying the current implementation is wrong. But at the same time, I can fully understand the confusion. I expected to see something too.

> Does it deserve an extra **Note** in the documentation maybe?

I think we should maybe have such note. It shouldn't hurt at least.

Thanks for the input @chrhol!
